### ISSUE TITLE
Pre-processing Steps & Census API Version

### ISSUE THREAD

**Issue by yfletberliac** [bowang-lab/scGPT#56]
Hi,

Firstly, congrats on a great paper and your work on this repo is very much appreciated. I'd to ask two questions:

1. Could you please clarify the pre-processing steps used in pre-training scGPT, i.e. did you use:

- Filtering by gene count?
- Filtering by cell count?
- Normalization?
- log1p transformation?
- Selection of highly variable genes?
- Binning (if used, what was the number of bins)?

2. The referenced `VERSION="2023-05-08"` doesn't seem to be a valid census API version, so we tried `VERSION="2023-05-15"` but we did not collect the same number of cells for each tissue. Could you please advise how to use `VERSION="2023-05-08"` or suggest an alternative version?

Thanks in advance!

Best,
Yannis

**Comment by subercui** [comment]
Hi @yfletberliac , thank you for the questions! Please let me first clarify the second question: we did use the `VERSION="2023-05-08"` for our data collection and pretraining. It appears the Census later remove the access for this version, although back then the version was listed as one of the LTS versions. I think this is because the Census is also a relatively new project and they are updating their versions quickly. To conclude, I think you may well use newer versions of the Census releases, they likely provide access to larger volumes of sequencing data now and which should contribute to even better pretraining.

**Comment by yfletberliac** [comment]
Thank you very much @subercui for your answer! That's helpful and I'll follow your reco.

Regarding the first part of my question, do you have an update on this, or maybe a pointer to the code where I can find the answer directly?

Best,
Yannis

**Comment by subercui** [comment]
Hi @yfletberliac , the pretraining code has not been fully merged, but you can find it in the `dev-temp` branch. I would like to point to this example sbatch script, https://github.com/bowang-lab/scGPT/blob/dev-temp/examples/sbatch_census.sh, which exemplifies the pretraining run by using the pretrain.py in the same folder.

To summarize, we did filter genes by https://github.com/bowang-lab/scGPT/blob/e2bd9d9958b89d8605a20eacc3ca3c4f57aaa68d/data/cellxgene/build_large_scale_data.py#L145-L147

then we simply trained on all genes with non-zero expression for each cell. A binning number of 51 is used by default in the training.

**Comment by yfletberliac** [comment]
Thanks @subercui for your answer!
